src/components/DocExplorer/Argument.js

```diff
@@ -0,0 +1,46 @@
+/**
+ *  Copyright (c) Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the license found in the
+ *  LICENSE file in the root directory of this source tree.
+ */
+
+import React, { PropTypes } from 'react';
+import TypeLink from './TypeLink';
+
+export default class Argument extends React.Component {
+  static propTypes = {
+    arg: PropTypes.object.isRequired,
+    onClickType: PropTypes.func.isRequired,
+  }
+
+  shouldComponentUpdate(nextProps) {
+    return this.props.arg !== nextProps.arg ||
```

Comment:
Wait, wouldn't `this.props.arg !== nextProps.arg` check all properties? Aren't the below checks redundant?